sigs/sig-serverless.md

```diff
@@ -0,0 +1,139 @@
+## CNCF Serverless SIG Charter
+
+## Introduction
+
+This is the charter referred to in “[CNCF
+SIGs](https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#sig-charter)”
+by the CNCF TOC, and consistent with the [proposed SIG
+definition](https://github.com/cncf/toc/blob/master/sigs/proposed.md).
+
+Borrowing from the [Serverless
+Whitepaper](https://github.com/cncf/wg-serverless/tree/master/whitepapers/serverless-overview),
+Serverless computing refers to the concept of building and running
+applications that do not require server management. Allowing developers
+to focus on writing their applications’ business logic rather than the
+complexities of the platform hosting/managing their applications. As such,
+this SIG will focus on technologies related to enabling this simplified
+developer experience.
+
+## Areas Considered In Scope
+
+Technologies related to hosting, managing, developing and integrating of
+Serverless applications from an application developer's perspective.
+The following are example areas that are considered in scope:
+
+* container hosting platforms that attempt to abstract their complexities away
```

Comment:
I suspect that we're going to have troubling being too crisp about the split between Serverless and Runtime (or App-Delivery), but to me the diff is similar to the diff between a Function author and a Pod author. In the end both just write code, and in some cases, both just end up running their app in pods. But, the environments are a bit different. One is focused on hiding the complexity of how things get hosted, while the other doesn't.

This SIG is focused on the stuff needed by the developers who do not want, or need, to know how stuff is hosted - they just want it to run.

Reply:
It's one of those in-between areas.

The relationship with SIG-Runtime I see here is with something like Knative or OpenFaaS. They are tools that make developers (end-users) not need to know about how things run or are hosted. But there is someone or something that needs to know about the inner workings of these tools and how they run say in production for the specific case of serverless or functions.  The pattern is different from regular workloads since it typically involves many short-lived pods/containers.

Reply:
I think we're basically on the same page w.r.t. what the end-user needs to know.
I'm just not sure about the "short-lived" comment - I think that's an impl detail because even with platforms like Knative, things can be long lived - if you ask nicely :-)

Reply:
I was thinking more of the Knative pattern equivalent to AWS Lambda where there's a default timeout of 3 seconds (and a max of 900 seconds) per function that runs in a short-lived VM like Firecracker in the backend. But sure, if someone runs their own Knative they could implement it in their own way and set the timeout limit to what they'd like. :-)

Reply:
Yup - to me, something like those timeouts, are artificial and are an implementation choice, not a hard-n-fast-rule about Serverless itself.